packages/svelte/src/internal/client/reactivity/sources.js

```diff
@@ -159,11 +167,17 @@ export function set(source, value) {
 			}
 		}
 
-		if (DEV) {
-			for (const effect of inspect_effects) {
+		if (DEV && inspect_effects.size > 0) {
+			const inspects = Array.from(inspect_effects);
+			if (current_effect === null) {
+				// Triggering an effect sync can tear the signal graph, so to avoid this we need
+				// to ensure the graph has been flushed before triggering any inspect effects.
+				// This is expensive, but given this is a DEV mode only feature, it should be fine
```

Comment:
Would be good to explain the if condition, hope I got that right
```suggestion
				// Only needed when there's currently no effect, and flushing with one present
				// could have other unintended consequences, like effects running out of order.
				// This is expensive, but given this is a DEV mode only feature, it should be fine
```